core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceTypeUtil.java

```diff
@@ -82,6 +95,69 @@
                     .put("0", false)
                     .build();
 
+    /**
+     * Try to get a public static method with a single {@link String} argument in clazz.
+     * @param clazz the class being introspected
+     * @param methodName the method expected
+     * @param <T> the expected result. It is *not* checked
+     * @return a function that takes a {@link String} and produces a {@link T}
+     */
+    @SuppressWarnings("unchecked")
+    @Nullable
+    private static <T> Function<String, T> getPublicStaticMethod(final Class<T> clazz,
+                                                                 final String methodName) {
+        try {
+            final Method method = clazz.getMethod(methodName, String.class);
+            return (str) -> {
+                try {
+                    return (T) method.invoke(null, str);
+                } catch (RuntimeException e) {
+                    throw e;
+                } catch (Exception e) {
+                    throw new IllegalArgumentException(e);
+                }
+            };
+        } catch (Throwable t) {
+            // No valid public static method found
+        }
+        return null;
+    }
+
+    @Nullable
+    private static <T> Function<String, T> getStringConstructor(final Class<T> clazz) {
+        try {
+            final Constructor<T> constructor = clazz.getConstructor(String.class);
+            return (str) -> {
+                try {
+                    return constructor.newInstance(str);
+                } catch (RuntimeException e) {
+                    throw e;
+                } catch (Exception e) {
+                    throw new IllegalArgumentException(e);
+                }
```

Comment:
- Could use `Exceptions.throwUnsafely()` instead of wrapping with `IAE`?
- Could catch an `InvocationTargetException` and unwrap before throwing

Reply:
Will do

Reply:
Fixed